conbench/entities/run.py

```diff
@@ -436,50 +356,59 @@ def _guts(cinfo: TypeCommitInfoGitHub) -> Tuple[Commit, bool]:
 
 
 class _Serializer(EntitySerializer):
-    def _dump(self, run: Run, get_baseline_runs: bool = False):
-        if run.commit:
-            commit = CommitSerializer().one.dump(run.commit)
-            commit.pop("links", None)
+    def _dump(
+        self, benchmark_result: "BenchmarkResult", get_baseline_runs: bool = False
+    ):
+        if benchmark_result.commit:
+            commit_dict = CommitSerializer().one.dump(benchmark_result.commit)
+            commit_dict.pop("links", None)
             commit_link = f.url_for(
-                "api.commit", commit_id=commit["id"], _external=True
+                "api.commit", commit_id=commit_dict["id"], _external=True
             )
         else:
-            commit = None
+            commit_dict = None
             commit_link = None
 
-        hardware = HardwareSerializer().one.dump(run.hardware)
-        hardware.pop("links", None)
-        result = {
-            "id": run.id,
-            "name": run.name,
-            "reason": run.reason,
+        hardware_dict = HardwareSerializer().one.dump(benchmark_result.hardware)
+        hardware_dict.pop("links", None)
+        out_dict = {
+            "id": benchmark_result.run_id,
+            "tags": benchmark_result.run_tags,
+            "reason": benchmark_result.run_reason,
+            "name": benchmark_result.run_tags.get("name"),
             "timestamp": conbench.util.tznaive_dt_to_aware_iso8601_for_api(
-                run.timestamp
-            ),
+                benchmark_result.run_tags["timestamp"]
```

Comment:
Maybe this should be the first /earliest-inserted benchmark result for this run ID (instead of a random / arbitrary one). Then this timestamp can be documented with 'when the first result came in'.

Reply:
Good idea. At least this will make the API result deterministic.